core/src/main/java/pt/webdetails/cda/push/WebsocketJsonQueryEndpoint.java

```diff
@@ -0,0 +1,225 @@
+/*!
+ * Copyright 2018 Webdetails, a Hitachi Vantara company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to  http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+
+package pt.webdetails.cda.push;
+
+import io.reactivex.BackpressureOverflowStrategy;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Observer;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.subjects.PublishSubject;
+import org.apache.commons.beanutils.BeanUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.pentaho.di.core.RowMetaAndData;
+import org.pentaho.platform.api.websocket.IWebsocketEndpoint;
```

Comment:
Yes, this should be on pentaho. That's exactly what will have a different implementation for foundry/osgi, so no need to be on core.

Reply:
Reference removed, by moving the interface into cda-core, and refactoring where it was being used in pentaho platform to reference the Class<?> instead of Class<? implements IWebsocketEndpoint>